clang/lib/Sema/SemaOpenMP.cpp

```diff
@@ -13004,13 +13004,34 @@ StmtResult SemaOpenMP::ActOnOpenMPTargetUpdateDirective(
                                           Clauses, AStmt);
 }
 
+// This checks whether num_teams clause only has one expression.
+static bool checkNumTeamsClauseSingleExpr(SemaBase &SemaRef,
```

Comment:
This function will be extended later to support `thread_limit` clause.

Reply:
Pass the maximal number of expressions as a parameter since you also need to check the ompx_bare version has at most 3. The diagnosis message then needs to print the maximum, maybe also mention ompx_bare.

Reply:
I'm thinking to emit a warning instead of an error. The warning saying only the first three expressions will be used. WDYT?